### PR TITLE
fix: request > limit fix | set max to limit between request and limit

### PR DESCRIPTION
**Pull request by ramantehlan**
> [!NOTE]
> **Low Risk**
> Low risk: small validation/clamping change to resource recommendation math, covered by new unit tests; main impact is slightly higher limits in edge cases.
> 
> **Overview**
> Prevents invalid recommendation applications by ensuring computed `memoryLimit` is clamped to at least `memoryRequest` (even when 7-day/OOM-derived limit would be lower) and `cpuLimit` is clamped to at least `cpuRequest` when a CPU limit is set.
> 
> Adds unit tests covering both edge cases to guard against Kubernetes `request <= limit` validation failures.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 5034a3f7fc08b938ed568d56d10d715828e5a4a0. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

## Summary by CodeRabbit

* **Bug Fixes**
  * Fixed resource recommendations to enforce Kubernetes validity constraints. Memory limits are now bounded to not fall below memory requests, and CPU limits are clamped to be at least CPU requests, ensuring `request ≤ limit` relationships.

* **Tests**
  * Added unit tests to verify proper enforcement of resource request-limit constraints.

### CODE REVIEW

File: pkg/task/utils/recommendation_apply.go [hunk 1]
```diff
@@ -103,14 +103,22 @@ func ComputeRecommendedResourceValues(ctx context.Context, rec PodContainerRecom
 			if containerStat.MemoryStats != nil && containerStat.MemoryStats.OOMMemory > 0 {
 				oom = containerStat.MemoryStats.OOMMemory
 			}
-			memoryLimit = EnforceMinimumMemory(max(memMax, oom) * 2)
+			// Keep derived memory limit from 7-day/OOM signal, but never let it
+			// drop below the chosen memory request, otherwise Kubernetes rejects
+			// the pod update (request must be <= limit).
+			memoryLimit = max(memoryRequest, EnforceMinimumMemory(max(memMax, oom)*2))
```
[the remark this answers is not in the document]
Comment (ramantehlan) [reply pkg/task/utils/recommendation_apply.go]: Will check

**Comment by ramantehlan** [comment]
Closing this for now, to debug the actual reason for this event. This PR silently ignores it.
